blockchain/ethereum/contracts compiler/internal/3cf6f504-22da-4d9e-aeb8-3e3547d397c8.json

```diff
@@ -2,7 +2,7 @@
: [\n                                \"evm.bytecode.object\",\n                                \"evm.deployedBytecode.object\",\n                                \"abi\",\n                                \"evm.bytecode.sourceMap\",\n                                \"evm.deployedBytecode.sourceMap\",\n                                \"metadata\"\n                            ],\n                            \"\": [\"ast\"]\n                        }\n                    },\n                    ...compileOptions\n                };\n\n                // Solidity Compiler's expected format\n                const input = {\n                    language: 'Solidity',\n                    sources: sources,\n                    settings: settings\n                };\n\n                // Parse the compiler output to retrieve the ABI and Bytecode\n                const output = solcSnapshot.compile(JSON.stringify(input), { import: findImports });\n                const outputParsed = JSON.parse(output);\n\n                if (hasError(outputParsed)){\n                    sendToOutput(\"error\", output, tag);\n                    shared.compilationRunning = false;\n                    return;\n                }\n\n                //  Ignore the intermediate compile results\n                const filtered = Object.fromEntries(Object.entries(outputParsed.contracts).filter(([key, _value]) => key.startsWith(rootPrefix)));\n                //  Elevate the results one level: use contract names as keys\n                compiledContracts = Object.entries(filtered).reduce((acc, [key, value]) => ({ ...acc, ...value }), {});\n\n                sendToOutput(\"compiled contracts\", compiledContracts, tag);\n            } else {\n                solcError = err;\n                sendToOutput(\"error\", err.toString(), tag);\n            }\n\n            shared.compilationRunning = false;\n        });\n    }\n}\n\nCompile();\n\nfunction sendToOutput(name, value, tag){\n    setTimeout(()=>{\n        outputs[name](value, tag);\n    }, 0);\n}\n\nfunction getContractByPath(path) {\n    return path.split('/').reduce((p, c) => p?.[c], contracts);\n}\n\n//  does not use node.js only path functions\nfunction splitPath(fullPath) {\n    const route = fullPath.split('/');\n    const file = route.pop();\n    const path = route.join('/');\n    return { file: file, path: path, fullPath: fullPath };\n}\n\n//  see:\n//  https://github.com/ethereum/solc-js/blob/master/README.md\n//  https://www.npmjs.com/package/solc4\nfunction findImports(path) {\n    for (const i in includeRoots) {\n        const fullPath = `${includeRoots[i]}/${path}`;\n        const res = getContractByPath(fullPath);\n        if (res) {\n            return {\n                contents: res\n            };\n        }\n    }\n}\n\nfunction hasError(outputParsed){\n    return (!outputParsed) || (outputParsed.errors && outputParsed.errors.some(\n      x => x.severity && x.severity.toLowerCase() === 'error'\n    ));\n  }\n\n"
```

Comment:
We usually sends out errors in a minimal structure of `{error: string}`
As far as I understand here we only send it out as `string`.
Otherwise I still experienced the cocurrency problems on my local Ganache installation that this change intended to fix, so not sure if it really meets its goal, but this should be double chacked.

Reply:
If I'm correct, the second part is has been confirmed meanwhile.
Submitted correction for the error output type.